### PR TITLE
Unvendor libgit2

### PR DESCRIPTION
**Pull request by xhochy**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/build.sh [hunk 1]
```diff
@@ -7,7 +7,8 @@ if [[ "${target_platform}" == "linux-64" ]]; then
   export RUSTFLAGS='-C target-feature=+fxsr,+sse,+sse2,+sse3,+ssse3,+sse4.1,+sse4.2,+popcnt,+avx,+fma'
 fi
 
-echo rustc --version
+export LIBGIT2_NO_VENDOR=1
+export CARGO_BUILD_RUSTFLAGS="${CARGO_BUILD_RUSTFLAGS} -C link-arg=-Wl,-L$PREFIX/lib"
```
[the remark this answers is not in the document]
Comment (0xbe7a) [reply recipe/build.sh:11]: Maturin might set [1] `CARGO_ENCODED_RUSTFLAGS`, which would result in cargo ignoring all other sources of `RUSTFLAGS` with lower priority including`CARGO_BUILD_RUSTFLAGS` [2]

[1] https://github.com/ravenexp/maturin/blob/3807c2bcdd9f49fdc6e8b5dc7de6ca1f879b1dfb/src/compile.rs
[2] https://doc.rust-lang.org/cargo/reference/config.html#buildrustflags

[the remark this answers is not in the document]
Comment (0xbe7a) [reply recipe/build.sh:11]: Can you set run `export RUST_LOG="warn,maturin=debug"` before invoking maturin through pip install?

[the remark this answers is not in the document]
Comment (xhochy) [reply recipe/build.sh:11]:
```suggestion
export CARGO_BUILD_RUSTFLAGS="${CARGO_BUILD_RUSTFLAGS} -C link-arg=-Wl,-L$PREFIX/lib"
export RUST_LOG="warn,maturin=debug"
```

[the remark this answers is not in the document]
Comment (0xbe7a) [reply recipe/build.sh:11]:
```
  2023-08-23T10:48:44.684090Z DEBUG maturin::compile: Running CARGO_ENCODED_RUSTFLAGS="-C\u{1f}link-arg=-Wl,-rpath,/Users/runner/miniforge3/conda-bld/polars_1692787235812/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib\u{1f}-C\u{1f}link-arg=-Wl,-headerpad_max_install_names\u{1f}-C\u{1f}link-arg=-Wl,-dead_strip_dylibs\u{1f}-C\u{1f}link-arg=-Wl,-L/Users/runner/miniforge3/conda-bld/polars_1692787235812/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib" PYO3_ENVIRONMENT_SIGNATURE="cpython-3.10-64bit" PYO3_PYTHON="/Users/runner/miniforge3/conda-bld/polars_1692787235812/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/bin/python" PYTHON_SYS_EXECUTABLE="/Users/runner/miniforge3/conda-bld/polars_1692787235812/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/bin/python" "cargo" "rustc" "--target" "x86_64-apple-darwin" "--message-format" "json-render-diagnostics" "--manifest-path" "/Users/runner/miniforge3/conda-bld/polars_1692787235812/work/Cargo.toml" "--release" "--lib" "--" "-C" "link-arg=-undefined" "-C" "link-arg=dynamic_lookup" "-C" "link-args=-Wl,-install_name,@rpath/polars.abi3.so"
```

CARGO_ENCODED_RUSTFLAGS LGTM 🤔

**Pull request closed without merging**